File: frontend/src/constants/ApiEndPoints.js
export class ApiEndpoints {
  // static BASE_URL = "http://localhost:8080/api/v1";
  static BASE_URL = "http://localhost:8080/api/v1";

  static user = {
    logIn: `${ApiEndpoints.BASE_URL}/users/login`,
    signUp: `${ApiEndpoints.BASE_URL}/users/signup`,
  };

  static team = {
    getTeam: `${ApiEndpoints.BASE_URL}/teams`,
    updateTeam: `${ApiEndpoints.BASE_URL}/teams`,
  };

  static player = {
    getPlayer: `${ApiEndpoints.BASE_URL}/players`,
    getPlayerCount: `${ApiEndpoints.BASE_URL}/players/count`,
    updatePlayer: `${ApiEndpoints.BASE_URL}/players`,
  };

  static transfer = {
    getTransferList: `${ApiEndpoints.BASE_URL}/transfers`,
    getTransferCount: `${ApiEndpoints.BASE_URL}/transfers/count`,
    createTransferList: `${ApiEndpoints.BASE_URL}/transfers`,
    applyTransfer: `${ApiEndpoints.BASE_URL}/transfers/{transferId}/apply`,
  };

  static util = {
    getAllCountry: `${ApiEndpoints.BASE_URL}/utils/countries`,
  };
}

File: backend/src/main/java/com/toptal/soccer/constants/ApiEndPoints.java
package com.toptal.soccer.constants;

public interface ApiEndPoints {
  String BASE_V1 = "/api/v1";

  // Controller base paths
  String HEALTH_BASE = BASE_V1 + "/health";
  String USERS_BASE = BASE_V1 + "/users";
  String TEAMS_BASE = BASE_V1 + "/teams";
  String PLAYERS_BASE = BASE_V1 + "/players";
  String TRANSFER_BASE = BASE_V1 + "/transfers";
  String UTILS_BASE = BASE_V1 + "/utils";

  // rest paths
  String USERS_LOGIN = "/login";
  String USERS_SIGNUP = "/signup";
  String COUNTS = "/count";
  String APPLY_TRANSFER = "/{transferId}/apply";

  static String getFullPath(String base, String path) {
    return (path == null) ? base : base + path;
  }
}
